### ISSUE TITLE
Bug: TypeError: FakeProducer.publish() got an unexpected keyword argument 'timeout'

### ISSUE THREAD

**Issue by Jonas1312** [airtai/faststream#1082]
Hi!

**Describe the bug**

My test does not pass. I get the error: `TypeError: FakeProducer.publish() got an unexpected keyword argument 'timeout'`

Traceback:

```text

    @override
    async def publish(  # type: ignore[override]
        self,
        message: SendableMessage = "",
        reply_to: str = "",
        correlation_id: Optional[str] = None,
        headers: Optional[Dict[str, str]] = None,
        **producer_kwargs: Any,
    ) -> Optional[DecodedMessage]:
        assert self._producer, NOT_CONNECTED_YET  # nosec B101
        assert self.subject, "You have to specify outgoing subject"  # nosec B101
    
        extra: AnyDict = {
            "reply_to": reply_to or self.reply_to,
        }
        if self.stream is not None:
            extra.update(
                {
                    "stream": self.stream.name,
                    "timeout": self.timeout,
                }
            )
    
>       return await self._producer.publish(
            message=message,
            subject=self.subject,
            headers=headers or self.headers,
            correlation_id=correlation_id,
            **extra,
            **producer_kwargs,
        )
E       TypeError: FakeProducer.publish() got an unexpected keyword argument 'timeout'

```

**How to reproduce**

```python
import uuid
from datetime import datetime
from typing import Literal

from faststream.nats import NatsBroker
from pydantic import BaseModel, Field

class NatsPublisher:

    def __init__(self, settings: NATSSettings, nats_broker: NatsBroker | None = None) -> None:
        """Initialize."""
        self._nats_broker = nats_broker or NatsBroker(settings.url)

        self._publisher_gen_started = self._nats_broker.publisher(
            "generation.started", stream=settings.stream
        )
        self._publisher_gen_ended = self._nats_broker.publisher(
            "generation.ended", stream=settings.stream
        )

    async def publish_event_generation_started(
        self, payload: GenerationStartedEventModel
    ) -> None:
        """Send generation started event."""
        await self._publisher_gen_started.publish(BaseMessage(content=payload.model_dump()))

    async def publish_event_generation_ended(
        self, payload: GenerationEndedEventModel
    ) -> None:
        """Send generation ended event."""
        await self._publisher_gen_ended.publish(BaseMessage(content=payload.model_dump()))

```

Test:
```python
import pytest
from faststream.nats import NatsBroker, TestNatsBroker

@pytest.mark.asyncio()
async def test_given_nats_publisher_when_publishing_gen_start_event_then_it_works():
    # Given
    nats_settings = NATSSettings()
    broker = NatsBroker()

    @broker.subscriber("generation.started")
    def some_subscriber(msg: BaseMessage) -> None:
        _ = msg

    some_message = "some_message"

    async with TestNatsBroker(broker) as br:
        nats_publisher = NatsPublisher(nats_settings, br)

        # When
        await nats_publisher.publish_event_generation_started(some_message)  

        # Then
        assert some_subscriber.mock
        some_subscriber.mock.assert_called_once_with(some_message)
```

**Environment**
Running FastStream 0.3.8 with CPython 3.11.5 on Darwin

If the code above is not good enough to find the bug, I can write a better minimal reproducible example.

Thanks!

**Comment by Lancetnik** [comment]
@Jonas1312 thx for the Issue! We'll upload a fix with `0.3.9`
Btw, Nats JS publisher `stream` option is not required. It just an extra check to validate that your subject is in correct stream, but regular `publish(subject=...)` works fine in JS case too

**Comment by Jonas1312** [comment]
> @Jonas1312 thx for the Issue! We'll upload a fix with `0.3.9` Btw, Nats JS publisher `stream` option is not required. It just an extra check to validate that your subject is in correct stream, but regular `publish(subject=...)` works fine in JS case too

I see, thanks a lot!
